packages/argo-admin-react/src/extension-api/data/examples/data.example.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import {extend, render, Text, useData} from '@shopify/argo-admin-react';
+
+function App() {
+  const {productId, variantId} = useData();
```

Comment:
We should be passing the extension point here for type safety
`const {productId, variantId} = useData<Admin::Product::SubscriptionPlan::Add>();`

Reply:
so far, I've been writing the examples using typescript but not actually doing any type annotations. this way, vscode catches obvious issues (like the fact that we had extensionPoint imported for every example) but the examples show up in 'JS'. probably should just have tsx and jsx versions for each though…

Reply:
For the `useData` and `useContainer` hooks you need to explicitly pass in the extension point otherwise the data object comes back as `any`. I don't think we want to encourage this? We want people to be able to get type safety for the data.

Reply:
yeah, we should just have JS and TS examples. i'll add them.

Can you get type safety using the typescript vanilla API?

Reply:
@iainmcampbell you get it for free in vanilla without doing anything. Only the hooks require manually specifying the extension point.

![image](https://user-images.githubusercontent.com/29458473/117071462-286f0580-acfd-11eb-8952-e1482823fbd2.png)

Reply:
ok so the way we have it in the template (define a named function, then pass that to `extend()` does require you to manually specify the extension point:

![Screen Shot 2021-05-05 at 1 57 11 PM](https://user-images.githubusercontent.com/3248903/117187382-d89a4800-ada9-11eb-85da-24158061fef2.png)

![Screen Shot 2021-05-05 at 1 57 23 PM](https://user-images.githubusercontent.com/3248903/117187387-d9cb7500-ada9-11eb-8842-c36a2b26dff0.png)

but if you pass an anonymous function to `extend()` you do get the types for free.
<img width="684" alt="Screen Shot 2021-05-05 at 1 57 58 PM" src="https://user-images.githubusercontent.com/3248903/117187434-e780fa80-ada9-11eb-96af-d8b6a784cd1e.png">